onnx/defs/traditionalml/defs.cc

```diff
@@ -915,131 +946,195 @@ ONNX_ML_OPERATOR_SET_SCHEMA(
     TreeEnsembleRegressor,
-    3,
+    5,
     OpSchema()
-        .SetDoc(TreeEnsembleRegressor_ver3_doc)
-        .Input(0, "X", "Input of shape [N,F]", "T")
-        .Output(0, "Y", "N classes", "tensor(float)")
-        .TypeConstraint(
-            "T",
-            {"tensor(float)", "tensor(double)", "tensor(int64)", "tensor(int32)"},
-            "The input type must be a tensor of a numeric type.")
-        .Attr("nodes_treeids", "Tree id for each node.", AttributeProto::INTS, OPTIONAL_VALUE)
+        .SetDoc(TreeEnsembleRegressor_ver5_doc)
+        .Input(0, "X", "Input of shape [Batch Size, Number of Features]", "T")
+        .Output(0, "Y", "Output of shape [Batch Size, Number of targets]", "T")
+        .TypeConstraint("T", {"tensor(float)", "tensor(double)"}, "The input type must be a tensor of a numeric type.")
+        .Attr("nodes_featureids", "Feature id for each node.", AttributeProto::INTS, true)
         .Attr(
-            "nodes_nodeids",
-            "Node id for each node. Node ids must restart at zero for each tree and increase sequentially.",
-            AttributeProto::INTS,
-            OPTIONAL_VALUE)
-        .Attr("nodes_featureids", "Feature id for each node.", AttributeProto::INTS, OPTIONAL_VALUE)
-        .Attr(
-            "nodes_values",
-            "Thresholds to do the splitting on for each node.",
-            AttributeProto::FLOATS,
-            OPTIONAL_VALUE)
-        .Attr(
-            "nodes_values_as_tensor",
-            "Thresholds to do the splitting on for each node.",
+            "nodes_splits",
+            "Thresholds to do the splitting on for each node with mode that is not 'SET_MEMBER'.",
             AttributeProto::TENSOR,
-            OPTIONAL_VALUE)
+            true)
         .Attr(
             "nodes_hitrates",
             "Popularity of each node, used for performance and may be omitted.",
-            AttributeProto::FLOATS,
-            OPTIONAL_VALUE)
-        .Attr(
-            "nodes_hitrates_as_tensor",
-            "Popularity of each node, used for performance and may be omitted.",
             AttributeProto::TENSOR,
             OPTIONAL_VALUE)
         .Attr(
             "nodes_modes",
-            "The node kind, that is, the comparison to make at the node. There is no comparison to make at a leaf node.<br>One of 'BRANCH_LEQ', 'BRANCH_LT', 'BRANCH_GTE', 'BRANCH_GT', 'BRANCH_EQ', 'BRANCH_NEQ', 'LEAF'",
-            AttributeProto::STRINGS,
-            OPTIONAL_VALUE)
-        .Attr("nodes_truenodeids", "Child node if expression is true", AttributeProto::INTS, OPTIONAL_VALUE)
-        .Attr("nodes_falsenodeids", "Child node if expression is false", AttributeProto::INTS, OPTIONAL_VALUE)
+            "The comparison operation performed by the node. This is encoded as an enumeration of 0 ('BRANCH_LEQ'), 1 "
+            "('BRANCH_LT'), 2 ('BRANCH_GTE'), 3 ('BRANCH_GT'), 4 ('BRANCH_EQ'), 5 ('BRANCH_NEQ'), and 6 "
+            "('SET_MEMBER').",
```

Comment:
Why SET_MEMBER instead of LEAF? It is not very intuitive.

Reply:
Leaves are not represented in any of the `nodes_*` attributes. See:
> The general encoding has been changed to reduce redundancy. Before, all nodes contained fields like truenodeids which are only relevant for interior nodes and not leaves. Since leaves will account for at least roughly half the nodes in a binary decision tree, this is highly wasteful. Therefore, this representation has fields for nodes_* for interior nodes and leaf_* for leaf nodes.

`SET_MEMBER` is a new mode. The leaf mode does not exist any more so it is not a replacement.

Reply:
Sorry, I changed many time my comment. I missed it was a new rule. I would prefer a name like BRANCH_MEMBER as all the other rules are BRANCH_ something.